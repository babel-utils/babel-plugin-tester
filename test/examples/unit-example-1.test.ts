/* eslint jest/require-hook: ["error", { "allowedFunctionCalls": ["pluginTester"] }] */

import path from 'node:path';

import { pluginTester } from 'universe';

import { identifierReversePlugin } from 'testverse:helpers/plugins.ts';

// * babel-plugin-tester example from
// * https://github.com/jamiebuilds/babel-handbook/blob/master/translations/nl/plugin-handbook.md#babel-plugin-tester
pluginTester({
  filepath: path.join(__dirname, '../file.js'),
  plugin: identifierReversePlugin,
  fixtures: 'fixtures/simple-reversed',
  tests: {
    'does not change code with no identifiers': "'hello';",
    'changes this code': {
      code: "var hello = 'hi';",
      output: "var olleh = 'hi';"
    },
    'using fixtures files': {
      fixture: 'fixtures/codeFixture.js',
      outputFixture: 'fixtures/outputFixture.js'
    },
    'using jest snapshots': {
      code: `
        function sayHi(person) {
          return 'Hello ' + person + '!'
        }
      `,
      snapshot: true
    }
  }
});
