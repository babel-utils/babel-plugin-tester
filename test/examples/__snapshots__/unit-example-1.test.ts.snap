// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 5. using jest snapshots: 5. using jest snapshots 1`] = `


function sayHi(person) {
  return 'Hello ' + person + '!'
}
      

      ↓ ↓ ↓ ↓ ↓ ↓

function iHyas(nosrep) {
  return 'Hello ' + nosrep + '!';
}

`;
